blueflood-http/src/main/java/com/rackspacecloud/blueflood/tracker/Tracker.java

```diff
@@ -40,12 +40,27 @@
     private static final String trackerName = String.format("com.rackspacecloud.blueflood.tracker:type=%s", Tracker.class.getSimpleName());
     private static final Pattern patternGetTid = Pattern.compile( "/v\\d+\\.\\d+/([^/]+)/.*" );
 
+    // Tracker is a singleton
+    private static Tracker instance = new Tracker();
+    private static boolean isRegistered = false;
+
     static Set tenantIds = new HashSet();
     static boolean isTrackingDelayedMetrics = false;
     static Set<String> metricNames = new HashSet<String>();
 
-    public Tracker() {
-        registerMBean();
+    // private constructor for singleton
+    private Tracker(){}
+
+    public static Tracker getInstance() {
+        register();
+        return instance;
+    }
+
+    public static void register() {
```

Comment:
I don't understand this and the getInstance() method. If you want to make this a singleton, then lets make all methods be instance method (i.e: not static). Just make the above ```public void register()``` and make the caller do this:
```
Tracker.getInstance().register()
```
Remember, it is easier to mock instances than static methods.